File: Ems_frontend/src/services/TeamService.js
import apiClient from "../apiConfig";

const BASE_URL = "/teams";

export const getAllTeams = (departmentId = '') => {
  const url = departmentId ? `${BASE_URL}?departmentId=${departmentId}&size=100` : `${BASE_URL}?size=100`;
  return apiClient.get(url);
};
export const createTeam = (team) => apiClient.post(BASE_URL, team);
export const getTeamById = (id) => apiClient.get(`${BASE_URL}/${id}`);
export const updateTeam = (id, team) => apiClient.put(`${BASE_URL}/${id}`, team);
export const deleteTeam = (id) => apiClient.delete(`${BASE_URL}/${id}`);

File: Ems_backend/src/main/java/com/ak/ems/service/TeamService.java
package com.ak.ems.service;

import com.ak.ems.dto.TeamDto;
import com.ak.ems.response.PageResponse;

public interface TeamService {
    TeamDto createTeam(TeamDto teamDto);
    TeamDto getTeamById(Long teamId);
    PageResponse<TeamDto> getAllTeams(String query, Long departmentId, int page, int size, String sortBy, String sortDir);
    TeamDto updateTeam(Long teamId, TeamDto teamDto);
    void deleteTeam(Long teamId);
}
